File: 06_pair_sum_exists.c
#include <stdio.h>

int pairSumExists(int arr[], int n, int target) {
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            if (arr[i] + arr[j] == target) return 1;
        }
    }
    return 0;
}

int main() {
    int arr[] = {10, 15, 3, 7};
    int n = 4, target = 17;
    printf("%d\n", pairSumExists(arr, n, target));
    return 0;
}

File: 06_pair_sum_exists.py
def pair_sum_exists(arr, target):
    seen = set()
    for num in arr:
        if target - num in seen:
            return True
        seen.add(num)
    return False

if __name__ == "__main__":
    arr = [10, 15, 3, 7]
    target = 17
    print(pair_sum_exists(arr, target))
